sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -30,46 +31,20 @@ class Logger final : public opentelemetry::logs::Logger
   explicit Logger(
       opentelemetry::nostd::string_view name,
       std::shared_ptr<LoggerContext> context,
-      std::unique_ptr<instrumentationscope::InstrumentationScope> instrumentation_scope =
+      nostd::unique_ptr<instrumentationscope::InstrumentationScope> instrumentation_scope =
```

Comment:
Why do we need to switch to `nostd::unique_ptr` here, and in general in most of other places in SDK ?

Reply:
Isn't it has better ABI compatibility? I think we should better use `nostd::unique_ptr` everywhere.

Reply:
We don't claim ABI compatibility at SDK level. And so have been using `std::` classes throughout SDK ( in both traces and metrics). Only API interfaces implemented by SDK use `nostd::` classes.

Reply:
Thanks for the explaination, I restore `nostd::unique_ptr` to `std::unique_ptr` in sdk and exporters now.